libs/langchain/langchain/retrievers/document_transformers/sumarize_transformer.py

```diff
@@ -0,0 +1,125 @@
+import asyncio
+from functools import partial
+from typing import Callable, Sequence, Optional, Dict, Any, Generator
+
+from langchain.chains import LLMChain
+from langchain.output_parsers import NumberedListOutputParser
+from langchain.prompts import PromptTemplate
+from langchain.schema import Document, BaseDocumentTransformer
+from langchain.schema.language_model import BaseLanguageModel
+
+
+def _default_get_input(doc: Document) -> Dict[str, Any]:
+    """Return the context chain input."""
+    return {
+        "context": doc.page_content,
+    }
+
+
+_default_template = """
+Sumarize a text input in the same language. 
+Context:
+```
+{context}
+```
+"""
+_default_format_instruction = NumberedListOutputParser()
+
+
+def _get_default_chain_prompt() -> PromptTemplate:
+    return PromptTemplate.from_template(
+        template=_default_template,
+    )
+
+
+class SummarizeTransformer(BaseDocumentTransformer):
+    """Generate questions for each Documents."""
+
+    def __init__(
+            self,
+            llm_chain: LLMChain,
```

Comment:
@pprados You may have seen LCEL already, but if not take a look at: https://python.langchain.com/docs/expression_language/

LLMChain is in a sense a legacy primitive right now. Users should generally use a prompt, an llm and a parser and chain them together with LCEL.

With an appropriate abstraction for the transformation pipeline, the summarize and generate question transformers should be reduced to only specifying an LLM, a prompt and declaring how one should execute the transformation (e.g., threading/processing/batch size etc.); there should be no implementation logic like `async def lazy_atransform_documents(` as that should be implemented only once in whatever entity is taking care fo the transformation pipeline.